src/main/java/org/terasology/web/resources/games/NewGameAction.java

```diff
@@ -0,0 +1,101 @@
+import org.terasology.module.Module;
+import org.terasology.module.ResolutionResult;
+import org.terasology.naming.Name;
+import org.terasology.web.io.ActionResult;
+import org.terasology.web.resources.ResourceAccessException;
+import org.terasology.world.internal.WorldInfo;
+import org.terasology.world.time.WorldTime;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import static org.terasology.web.resources.InputCheckUtils.checkNotNull;
+import static org.terasology.web.resources.InputCheckUtils.checkNotNullOrEmpty;
+
+public class NewGameAction extends AbstractAction {
+
+    private String gameName;
+    private String seed;
+    private List<Name> modules;
+    private SimpleUri worldGenerator;
+
+    private transient DependencyResolver dependencyResolver;
+
+    @Override
+    public void perform(ModuleManager moduleManager) throws ResourceAccessException {
+        checkNotNullOrEmpty(gameName, "A name for the new game must be specified.");
+        checkNotNullOrEmpty(seed, "A seed must be specified.");
+        checkNotNull(modules, "A list of modules must be specified");
+        checkNotNull(worldGenerator, "A world generator must be specified.");
+        if (Files.exists(getPathManager().getSavePath(gameName))) {
+            throw new ResourceAccessException(new ActionResult(ActionResult.Status.CONFLICT, "A game with the specified name already exists"));
+        }
+        if (dependencyResolver == null) {
+            dependencyResolver = new DependencyResolver(moduleManager.getRegistry());
+        }
+        GameManifest newGameManifest = buildGameManifest();
+        Path saveDir = getPathManager().getSavePath(newGameManifest.getTitle());
+        Path saveFile = saveDir.resolve(GameManifest.DEFAULT_FILE_NAME);
+        try {
+            Files.createDirectory(saveDir);
+            GameManifest.save(saveFile, newGameManifest);
+        } catch (IOException ex) {
+            throw new ResourceAccessException(new ActionResult(ActionResult.Status.GENERIC_ERROR, "Failed to save new game manifest: " + ex.getMessage()));
+        }
+    }
+
+    private GameManifest buildGameManifest() throws ResourceAccessException {
+        GameManifest newGameManifest = new GameManifest();
+        newGameManifest.setTitle(gameName);
+        newGameManifest.setSeed(seed);
+
+        ResolutionResult result = dependencyResolver.resolve(modules);
+        if (!result.isSuccess()) {
+            throw new ResourceAccessException(new ActionResult(ActionResult.Status.GENERIC_ERROR, "Failed to resolve all module dependencies"));
+        }
+        for (Module m: result.getModules()) {
+            newGameManifest.addModule(m.getId(), m.getVersion());
+        }
+
+        WorldInfo worldInfo = new WorldInfo(TerasologyConstants.MAIN_WORLD, newGameManifest.getSeed(),
+                (long) (WorldTime.DAY_LENGTH * 0.025f), worldGenerator);
```

Comment:
Yes, these 2 lines are taken from [StateHeadlessSetup](https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/java/org/terasology/engine/subsystem/headless/mode/StateHeadlessSetup.java#L141); the time offset is set to that value to avoid the player spawning in a completely dark environment. In my understanding though it's not possible to reuse the whole [createGameManifest](https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/java/org/terasology/engine/subsystem/headless/mode/StateHeadlessSetup.java#L109) method since it just uses the default settings for title, seed and world generator while here we are using user-supplied values.

As you suggest, in another branch I could implement a Resource to allow to view and/or manipulate the world time of an existing savegame or maybe the currently running game. In my understanding I still need to set the world time to some value when initializing the game manifest, I could set it to zero if you prefer?